src/smc-webapp/jupyter/actions.ts

```diff
@@ -29,15 +29,26 @@ import {
   format_parser_for_extension
 } from "../../smc-util/code-formatter";
 
-import { Actions } from "../app-framework";
+const misc = require("smc-util/misc");
+const { required, defaults } = misc;
+import { Actions, AppRedux } from "../app-framework";
 import {
   JupyterStoreState,
   JupyterStore,
   show_kernel_selector_reasons
 } from "./store";
+const util = require("./util");
+const parsing = require("./parsing");
+const keyboard = require("./keyboard");
+import * as nbgrader from "./nbgrader";
+const cell_utils = require("./cell-utils");
+const { cm_options } = require("./cm_options");
+const { JUPYTER_CLASSIC_MODERN } = require("smc-util/theme");
+=======
 import * as parsing from "./parsing";
 import * as cell_utils from "./cell-utils";
 import { cm_options } from "./cm_options";
+>>>>>>> origin/master
```

Comment:
Just a comment - this is a scary bad merge.  For example, you left in the >>>>'s.  Also, it adds back in stuff like
```
const misc = require("smc-util/misc");
const { required, defaults } = misc;
```
that I put a lot of work into removing (so as to get more typescript value).  

**No action is required -- since I may just look at your code for inspiration.**